### PR TITLE
CDAP-3115: cube sink docs

### PR DESCRIPTION
**Pull request by abaranau**

### CODE REVIEW

File: cdap-docs/application-templates/source/etl/templates/sinks/batch/cube.rst [hunk 1]
```diff
@@ -8,7 +8,9 @@ Sinks: Batch: Cube
 
 .. rubric:: Description
 
-A BatchCubeSink that takes a StructuredRecord in, maps it to a CubeFact, and writes it to
+Batch sink that writes to a Cube dataset.
+
+BatchCubeSink takes a StructuredRecord in, maps it to a CubeFact, and writes it to
 the Cube dataset identified by the name property.
 
 If Cube dataset does not exist, it will be created using properties provided with this
```
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sinks/batch/cube.rst]: If the Cube dataset...

File: cdap-docs/application-templates/source/etl/templates/sinks/batch/cube.rst [hunk 2]
```diff
@@ -16,7 +18,9 @@ sink.
 
 .. rubric:: Use Case
 
-TODO: Fill me out
+A BatchCubeSink is used to write data into a Cube from batch-enabled data sources to allow further data analysis.
+E.g. with BatchCubeSink you can periodically upload data from PartitionedFileSet into a Cube to perform complex
```
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sinks/batch/cube.rst]: For example, you can periodically upload data from a PartitionedFileSet into a Cube to perform complex

File: cdap-docs/application-templates/source/etl/templates/sinks/batch/cube.rst [hunk 3]
```diff
@@ -42,9 +46,78 @@ cubeFact.timestamp.field is provided).
 **cubeFact.measurements:** Measurements to be extracted from StructuredRecord to be used in CubeFact.
 Provide properties as a JSON Map. For example, to use the 'price' field as a measurement of type gauge,
 and the 'count' field as a measurement of type counter, the property should have the value:
```
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sinks/batch/cube.rst]: and the 'quantity' field

File: cdap-docs/application-templates/source/etl/templates/sinks/batch/cube.rst [hunk 4]
```diff
@@ -42,9 +46,78 @@ cubeFact.timestamp.field is provided).
...
-{"cubeFact.measurement.price":"GAUGE", "cubeFact.measurement.count":"COUNTER"}.
+{"cubeFact.measurement.price":"GAUGE", "cubeFact.measurement.quantity":"COUNTER"}.
 
 .. rubric:: Example
 
-TODO: Fill me out
-
+The following configuration tells to write data into Cube dataset with name "myCube"; provides dataset properties
+(dataset.*) for creating new dataset, if the one with given name doesn't exist; configures measurements to aggregate and
+how to determine timestamp for the facts to be written into Cube::
```
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sinks/batch/cube.rst]: This configuration specifies writing data into a Cube dataset named "myCube"; it provides dataset properties
(dataset.*) for creating new a dataset, if one with the given name doesn't exist; it then configures measurements to aggregate and specifies the timestamp format for facts being written into the Cube::

File: cdap-docs/application-templates/source/etl/templates/sinks/batch/cube.rst [hunk 5]
```diff
@@ -42,9 +46,78 @@ cubeFact.timestamp.field is provided).
...
+
+    {
+        "name": "myCube",
+
+        "dataset.cube.resolutions": "1,60,3600",
+        "dataset.cube.properties": {
+          "dataset.cube.aggregation.byName.dimensions": "name",
+          "dataset.cube.aggregation.byNameByZip.dimensions": "name,zip",
+        },
+
```
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sinks/batch/cube.rst]: remove blank line

File: cdap-docs/application-templates/source/etl/templates/sinks/batch/cube.rst [hunk 6]
```diff
@@ -42,9 +46,78 @@ cubeFact.timestamp.field is provided).
...
 
-TODO: Fill me out
-
+The following configuration tells to write data into Cube dataset with name "myCube"; provides dataset properties
+(dataset.*) for creating new dataset, if the one with given name doesn't exist; configures measurements to aggregate and
+how to determine timestamp for the facts to be written into Cube::
+
+    {
+        "name": "myCube",
+
```
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sinks/batch/cube.rst]: remove blank line

File: cdap-docs/application-templates/source/etl/templates/sinks/batch/cube.rst [hunk 7]
```diff
@@ -42,9 +46,78 @@ cubeFact.timestamp.field is provided).
...
+        "cubeFact.timestamp.field": "ts",
+        "cubeFact.timestamp.format": "MM/dd/yyyy HH:mm:ss",
+        "cubeFact.measurements": {
+          "cubeFact.measurement.price": "GAUGE"
+          "cubeFact.measurement.quantity": "COUNTER"
+        }
+    }
+
+Once data is there, you can now run queries using AbstractCubeHttpHandler (you will need to deploy an application that
```
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sinks/batch/cube.rst]: Once the data is there, you could run queries using an AbstractCubeHttpHandler. (You'd need to deploy an application
containing a service using it.) A sample query::

File: cdap-docs/application-templates/source/etl/templates/sinks/real-time/cube.rst [hunk 8]
```diff
@@ -8,17 +8,19 @@ Sinks: Real-time: Cube
 
 .. rubric:: Description
 
-Real-time Sink that writes data to a Cube dataset
+Real-time Sink that writes data to a Cube dataset.
 
-This RealtimeCubeSink takes a StructuredRecord in, maps it to a CubeFact, and writes it to
-a Cube dataset identified by the Dataset Name property.
+RealtimeCubeSink takes a StructuredRecord in, maps it to a CubeFact, and writes it to
```
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sinks/real-time/cube.rst]: RealtimeCubeSink takes in a StructuredRecord,

File: cdap-docs/application-templates/source/etl/templates/sinks/real-time/cube.rst [hunk 9]
```diff
@@ -8,17 +8,19 @@ Sinks: Real-time: Cube
...
+the Cube dataset identified by the name property.
 
 If Cube dataset does not exist, it will be created using properties provided with this
 sink.
 
 .. rubric:: Use Case
 
-TODO: Fill me out
+A RealtimeCubeSink is used to write data into a Cube from data sources to allow further OLAP data analysis.
+E.g. with RealtimeCubeSink you can write data from a Stream into a Cube as it arrives to perform complex
```
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sinks/real-time/cube.rst]: For example, you can write data from a Stream into a Cube as the data arrives to perform complex

File: cdap-docs/application-templates/source/etl/templates/sinks/real-time/cube.rst [hunk 10]
```diff
@@ -44,8 +46,78 @@ cubeFact.timestamp.field is provided).
 **cubeFact.measurements:** Measurements to be extracted from StructuredRecord to be used in CubeFact.
 Provide properties as a JSON Map. For example, to use the 'price' field as a measurement of type gauge,
 and the 'count' field as a measurement of type counter, the property should have the value:
```
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sinks/real-time/cube.rst]: and the 'quantity'

File: cdap-docs/application-templates/source/etl/templates/sinks/real-time/cube.rst [hunk 11]
```diff
@@ -44,8 +46,78 @@ cubeFact.timestamp.field is provided).
...
-{"cubeFact.measurement.price":"GAUGE", "cubeFact.measurement.count":"COUNTER"}.
+{"cubeFact.measurement.price":"GAUGE", "cubeFact.measurement.quantity":"COUNTER"}.
 
 .. rubric:: Example
 
-TODO: Fill me out
+The following configuration tells to write data into Cube dataset with name "myCube"; provides dataset properties
+(dataset.*) for creating new dataset, if the one with given name doesn't exist; configures measurements to aggregate and
+how to determine timestamp for the facts to be written into Cube::
```
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sinks/real-time/cube.rst]: This configuration specifies writing data into a Cube dataset named "myCube"; it provides dataset properties
(dataset.*) for creating a new dataset, if one with the given name doesn't exist; it then configures measurements to aggregate and specifies the timestamp format for facts being written into the Cube::

File: cdap-docs/application-templates/source/etl/templates/sinks/real-time/cube.rst [hunk 12]
```diff
@@ -44,8 +46,78 @@ cubeFact.timestamp.field is provided).
...
+
+    {
+        "name": "myCube",
+
```
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sinks/real-time/cube.rst]: remove blank line

File: cdap-docs/application-templates/source/etl/templates/sinks/real-time/cube.rst [hunk 13]
```diff
@@ -44,8 +46,78 @@ cubeFact.timestamp.field is provided).
...
+        "dataset.cube.resolutions": "1,60,3600",
+        "dataset.cube.properties": {
+          "dataset.cube.aggregation.byName.dimensions": "name",
+          "dataset.cube.aggregation.byNameByZip.dimensions": "name,zip",
+        },
+
```
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sinks/real-time/cube.rst]: remove blank line

File: cdap-docs/application-templates/source/etl/templates/sinks/real-time/cube.rst [hunk 14]
```diff
@@ -44,8 +46,78 @@ cubeFact.timestamp.field is provided).
...
+        "cubeFact.timestamp.field": "ts",
+        "cubeFact.timestamp.format": "MM/dd/yyyy HH:mm:ss",
+        "cubeFact.measurements": {
+          "cubeFact.measurement.price": "GAUGE"
+          "cubeFact.measurement.quantity": "COUNTER"
+        }
+    }
+
+Once data is there, you can now run queries using AbstractCubeHttpHandler (you will need to deploy an application that
+contains service using it), e.g.::
```
Comment (awholegunch) [inline cdap-docs/application-templates/source/etl/templates/sinks/real-time/cube.rst]: Once the data is there, you could run queries using an AbstractCubeHttpHandler. (You'd need to deploy an application
containing a service using it.) A sample query::

**Comment by awholegunch** [comment]
I've finished reviewing. Let me know if you'd like me to look again.

**Comment by abaranau** [comment]
@awholegunch could you please just edit in place? Totally trust you

**Comment by awholegunch** [comment]
Edits done. LGTM.

**Pull request merged**